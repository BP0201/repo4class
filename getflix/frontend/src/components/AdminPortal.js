import React, { useState, useEffect } from 'react';

const AdminPortal = ({ getAllUsers }) => {
  const [users, setUsers] = useState([]);

  useEffect(function loadUsers() {
    loadAllUsers()
  }, [])

  async function loadAllUsers() {
    // defined function from App, fetches users from backend
    setUsers(await getAllUsers());
  }

  return (
    <div className='AdminPortal container'>
      <h1 className='display-4'>Admin Portal</h1>
      <h3>All Users:</h3>
        <ul data-testid="ap-user-list">
          {users.map(u => (
            <li key={u.username}>
              {u.username}
            </li>
          ))}
        </ul>
    </div>
  )
}

export default AdminPortal;