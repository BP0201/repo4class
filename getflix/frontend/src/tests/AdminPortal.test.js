import React from "react";
import { render, fireEvent, cleanup } from '@testing-library/react';
import AdminPortal from '../components/AdminPortal';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from './testUtils';

let calledFn = null;

function mockFn() {
  calledFn = true
}

describe("AdminPortal", () => {
  it("matches snapshot", async () => {
      const { asFragment, getByTestId } = render(
        <MemoryRouter>
          <UserProvider>
            <AdminPortal getAllUsers={mockFn} />
          </UserProvider>
        </MemoryRouter>
      );
      expect(asFragment()).toMatchSnapshot();
      const userList = getByTestId('ap-user-list');
      expect(userList).toBeInTheDocument();
      expect(calledFn).toBeTruthy();
  });
});