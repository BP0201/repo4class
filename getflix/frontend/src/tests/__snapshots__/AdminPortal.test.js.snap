// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPortal matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="AdminPortal container"
  >
    <h1
      class="display-4"
    >
      Admin Portal
    </h1>
    <h3>
      All Users:
    </h3>
    <ul
      data-testid="ap-user-list"
    />
  </div>
</DocumentFragment>
`;
